src/Parser/RecursiveTextProcessor.php

```diff
@@ -295,6 +295,11 @@ private function pruneCategory( &$text ) {
 
 	private function getParserOutputSafe(): ?ParserOutput {
 		try {
+			// The getOutput() method is deprecated in MW 1.42 before initializing
+			// So we need to check if the parser was initialized.
+			if ( version_compare( MW_VERSION, '1.42', '>=' ) && $this->parser->getOptions() === null ) {
+				$this->parser->resetOutput();
```

Comment:
Does this work for you? When I tried this for another extension, investigating a problem, it failed on $this->mOptions I think? As it wasn't defined.

Reply:
Oh we’re checking if getOptions is null and if it is we are calling resetOutput. That will fail because of https://github.com/wikimedia/mediawiki/blob/b3f3e47f61ad544241319cb292b27e8acfcec13c/includes/parser/Parser.php#L648.

this should be reverted I think?

Reply:
@alistair3149

Reply:
That should be caught by the try block and return null. Is there an alternative way to initialize parser output?

Reply:
I’m not aware of any alternative.

Reply:
I don't understand why this is reverted, the fix was nested in a try/catch, so the original functionality (what it did before) stayed the same. 

The class is called with a parser 99th of the time, and if it doesn't, the parser is initialized with `MediaWikiServices::getInstance()->getParser()` which is causing the notice to show up (hence the resetOutput fix).

Reply:
For future reference, could we please wait a bit before reverting changes? I’d appreciate a chance to respond or discuss any concerns first. Thank you for understanding.

Reply:
You were checking if getOptions is null and if it was, calling resetOutput . Which would fail on this https://github.com/wikimedia/mediawiki/blob/b3f3e47f61ad544241319cb292b27e8acfcec13c/includes/parser/Parser.php#L648 line?

I realise now that the catch would make sure it wouldn’t fatal so sorry.

Reply:
Yeah your right about that. The fix is slightly different indeed, the function should return false if the options we're not set, since its an invalid state. I'll update the code (and comments) accordingly.